### DISCUSSION TITLE
Problem with merge function

### DISCUSSION THREAD

**Discussion by ljouneau** [theislab/scvelo#660] [Q&A]
Hello,

After having loaded 2 velocyto loom files, I tried to merge them but the result shows very few (22) observations 
>>> import scvelo as scv
>>> adata_t0 = scv.read("../Analyses_velocyto/icare5_t0h/velocyto/icare5_t0h.loom",cache=True)
>>> adata_t4 = scv.read("../Analyses_velocyto/icare5_t4h/velocyto/icare5_t4h.loom",cache=True)
>>> adata=scv.utils.merge(adata_t0,adata_t4)
>>> adata_t0
AnnData object with n_obs × n_vars = **8288** × 60672
    var: 'Accession', 'Chromosome', 'End', 'Start', 'Strand'
    layers: 'ambiguous', 'matrix', 'spliced', 'unspliced'
>>> adata_t4
AnnData object with n_obs × n_vars = **7841** × 60672
    obs: 'initial_size_spliced', 'initial_size_unspliced', 'initial_size'
    var: 'Accession', 'Chromosome', 'End', 'Start', 'Strand'
    layers: 'ambiguous', 'matrix', 'spliced', 'unspliced'
>>> adata
AnnData object with n_obs × n_vars = **22** × 60672
    obs: 'initial_size_spliced', 'initial_size_unspliced', 'initial_size'
    var: 'Accession', 'Chromosome', 'End', 'Start', 'Strand'
    layers: 'ambiguous', 'matrix', 'spliced', 'unspliced'

Thank you for your answers

**Comment by WeilerP** [comment]
Not sure what the question is but I presume it's why there are only 22 observations left after merging. Looks like your two AnnData objects only have 22 observations in common. You can check the number of common observations yourself by

```python
import pandas as pd
import scvelo as scv

adata_t0 = scv.read("../Analyses_velocyto/icare5_t0h/velocyto/icare5_t0h.loom",cache=True)
adata_t4 = scv.read("../Analyses_velocyto/icare5_t4h/velocyto/icare5_t4h.loom",cache=True)

pd.unique(adata_t0.obs_names.intersection(adata_t4.obs_names)).size
```
